File: python/detect.py
# -*- coding: utf-8 -*-

def getToken():
    ak = '官方获取的ak'
    sk = '官方获取的sk'
    host = f'https://aip.baidubce.com/oauth/2.0/token?grant_type=client_credentials&client_id={ak}&client_secret={sk}'
    response = requests.get(host)
    return response.json().get("access_token")

def img_to_base64(file_path):
    with open(file_path, 'rb') as f:
        base_64_data = base64.b64encode(f.read())
        s = base_64_data.decode()
        return s

def FaceDetect(token_, base_64_data):
    params = {}
    request_url = "https://aip.baidubce.com/rest/2.0/face/v3/detect"
    params["image"] = base_64_data
    params["image_type"] = "BASE64"
    params["face_field"] = "age,beauty"
    access_token = token_
    request_url = request_url + "?access_token=" + access_token
    headers = {'content-type': 'application/json'}
    response = requests.post(request_url, data=params, headers=headers)
    if response:
        print(response.json())
        print(response.json()["result"]["face_list"][0]["age"])
        print(response.json()["result"]["face_list"][0]["beauty"])


import requests
import base64
import json
if __name__ == "__main__":
    base_64 = img_to_base64("zhangsan.jpg")
    token  = getToken()
    FaceDetect(token, base_64)

File: java/src/main/java/api/baidu/detect/Detect.java
package api.baidu.detect;

import java.util.*;

import api.baidu.util.GetToken;
import api.baidu.util.GsonUtils;
import api.baidu.util.HttpUtil;

/**
* 人脸检测与属性分析
*/
public class Detect {

    /**
    * 重要提示代码中所需工具类
    * FileUtil,Base64Util,HttpUtil,GsonUtils请从
    * https://ai.baidu.com/file/658A35ABAB2D404FBF903F64D47C1F72
    * https://ai.baidu.com/file/C8D81F3301E24D2892968F09AE1AD6E2
    * https://ai.baidu.com/file/544D677F5D4E4F17B4122FBD60DB82B3
    * https://ai.baidu.com/file/470B3ACCA3FE43788B5A963BF0B625F3
    * 下载
    */
    public static String faceDetect(String base64) {
        // 请求url
        String url = "https://aip.baidubce.com/rest/2.0/face/v3/detect";
        try {
        	
            Map<String, Object> map = new HashMap<>();
            map.put("image", base64);
            map.put("face_field", "age,beauty,face_shape,gender,emotion,face_type");
            map.put("image_type", "BASE64");

            String param = GsonUtils.toJson(map);

            // 注意这里仅为了简化编码每一次请求都去获取access_token，线上环境access_token有过期时间， 客户端可自行缓存，过期后重新获取。
            String accessToken = GetToken.getAuth("recognise");

            String result = HttpUtil.post(url, accessToken, "application/json", param);
            return result;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

}